src/compiler/interface.jl

```diff
@@ -68,15 +68,20 @@ julia> gradient([7, 11], 0, 1) do x, y, d
          p = size(x, d)
          sum(x.^p .+ y)
        end
-([14.0, 22.0], 2, nothing)
+([14.0, 22.0], 2.0, nothing)
 ```
 """
 function gradient(f, args...)
   y, back = pullback(f, args...)
-  return back(sensitivity(y))
+  grad = back(sensitivity(y))
+  isnothing(grad) ? nothing : map(_project, args, grad)
 end
```

Comment:
You can add a method to `_project` and avoid this change

Reply:
> You can add a method to _project and avoid this change

Can you write exactly what method that would be?

Reply:
Something like `_project(x, ::Nothing) = nothing` maybe

Reply:
This is easy to try:
```
julia> _project(x, ::Nothing) = nothing
_project (generic function with 1 method)

julia> map(_project, (1,2,3), nothing)
ERROR: MethodError: no method matching length(::Nothing)
```